### ISSUE TITLE
nvidia 32-bit OpenGL support dropped as of version 460.x

### ISSUE THREAD

**Issue by foley2k2** [ValveSoftware/steam-for-linux#7865]
TLDR;
Due to not having a 64-bit Steam client, nvidia's driver has broken the store.

Symptoms:
run steam.sh in the terminal
never displays graphics

Workaround is supposedly the 450 build, will try that, but it's a pain since you have to kill X to update from the shell and ubuntu auto-restarts it unless you figure out the right service to stop.

all the gory details follow:

#### Your system information
$ inxi -F
System:    Host: jason-Linux Kernel: 5.4.0-74-generic x86_64 bits: 64 Desktop: Cinnamon 4.6.7 
           Distro: Linux Mint 20 Ulyana 
Machine:   Type: Desktop Mobo: MSI model: Z170A SLI (MS-7998) v: 1.0 serial: <superuser/root required> 
           UEFI [Legacy]: American Megatrends v: 3.D0 date: 06/15/2018 
CPU:       Topology: Quad Core model: Intel Core i5-7600K bits: 64 type: MCP L2 cache: 6144 KiB 
           Speed: 801 MHz min/max: 800/4200 MHz Core speeds (MHz): 1: 800 2: 800 3: 800 4: 800 
Graphics:  Device-1: NVIDIA GP104 [GeForce GTX 1080] driver: nvidia v: 460.80 
           Display: x11 server: X.Org 1.20.9 driver: nvidia 
           resolution: 1920x1080~60Hz, 1920x1080~60Hz, 1920x1080~60Hz 
           OpenGL: renderer: N/A v: N/A 
Audio:     Device-1: Intel 100 Series/C230 Series Family HD Audio driver: snd_hda_intel 
           Device-2: NVIDIA GP104 High Definition Audio driver: snd_hda_intel 
           Device-3: Logitech C922 Pro Stream Webcam type: USB driver: snd-usb-audio,uvcvideo 
           Sound Server: ALSA v: k5.4.0-74-generic 
Network:   Device-1: Intel Ethernet I219-V driver: e1000e 
           IF: enp0s31f6 state: up speed: 1000 Mbps duplex: full mac: 4c:cc:6a:f5:2c:88 
           IF-ID-1: docker0 state: down mac: 02:42:6c:21:d9:aa 
Drives:    Local Storage: total: 9.10 TiB used: 4.17 TiB (45.8%) 
           ID-1: /dev/nvme0n1 vendor: Mushkin model: MKNSSDHL1TB-D8 size: 931.51 GiB 
           ID-2: /dev/sda vendor: Western Digital model: WD1002FAEX-00Z3A0 size: 931.51 GiB 
           ID-3: /dev/sdb type: USB vendor: Western Digital model: WD My Book 25EE size: 7.28 TiB 
Partition: ID-1: / size: 915.40 GiB used: 672.94 GiB (73.5%) fs: ext4 dev: /dev/nvme0n1p5 
Sensors:   System Temperatures: cpu: 41.0 C mobo: 29.8 C gpu: nvidia temp: 54 C 
           Fan Speeds (RPM): N/A gpu: nvidia fan: 36% 
Info:      Processes: 264 Uptime: 50m Memory: 31.32 GiB used: 5.18 GiB (16.5%) Shell: bash inxi: 3.0.3

* Steam client version (build number or date): see below, exact build referenced
* Distribution (e.g. Ubuntu): Linux Mint 20
* Opted into Steam client beta?: can't check on this PC
* Have you checked for system updates?: all updates installed

#### Please describe your issue in as much detail as possible:
[2021-06-16 17:13:13] Checking for update on startup
[2021-06-16 17:13:13] Checking for available updates...
[2021-06-16 17:13:13] Downloading manifest: https://cdn.cloudflare.steamstatic.com/client/steam_client_ubuntu12
Installing breakpad exception handler for appid(steam)/version(1623193086)
[2021-06-16 17:13:13] Download skipped: /client/steam_client_ubuntu12 version 1623193086, installed version 1623193086, existing pending version 0
[2021-06-16 17:13:13] Nothing to do
[2021-06-16 17:13:13] Verifying installation...
[2021-06-16 17:13:13] Performing checksum verification of executable files
[2021-06-16 17:13:14] Verification complete
Loaded SDL version 2.0.15-6501165
Gtk-Message: 17:13:14.361: Failed to load module "gail"
Gtk-Message: 17:13:14.361: Failed to load module "atk-bridge"

(steam:12140): Gtk-WARNING **: 17:13:14.366: Unable to locate theme engine in module_path: "adwaita",
/usr/share/themes/Mint-Y/gtk-2.0/main.rc:1085: error: unexpected identifier 'direction', expected character '}'
libGL error: No matching fbConfigs or visuals found
libGL error: failed to load driver: swrast
Steam: An X Error occurred
X Error of failed request:  GLXBadContext
Major opcode of failed request:  151
Serial number of failed request:  52
xerror_handler: X failed, continuing
Steam: An X Error occurred
X Error of failed request:  BadValue (integer parameter out of range for operation)
Major opcode of failed request:  151
Value in failed request:  0x0
Serial number of failed request:  51
xerror_handler: X failed, continuing
Steam: An X Error occurred
X Error of failed request:  BadMatch (invalid parameter attributes)
Major opcode of failed request:  151
Serial number of failed request:  53
xerror_handler: X failed, continuing
assert_20210616171313_1.dmp[12172]: Uploading dump (out-of-process)
/tmp/dumps/assert_20210616171313_1.dmp
./steam.sh: line 772: 12140 Segmentation fault      (core dumped) $STEAM_DEBUGGER $DEBUGGER_ARGS "$STEAMROOT/$STEAMEXEPATH" "$@"

minidumpfile: [https://drive.google.com/file/d/1MNwqGgnmzZei-s1057X2eQmStLY7Oiar/view?usp=sharing](url)
(drag/drop filters .dmp files)

#### Steps for reproducing this issue:

1. install the latest nvidia Linux drivers
2. try to run steam from the GUI - nothing
3. try to run steam from the shell - you'll get the above

**Comment by kisak-valve** [comment]
Hello @foley2k2, your claim that the nVidia 460 series driver does not support 32 bit applications is not true.

Please share the output of `apt policy libnvidia-gl-460 libnvidia-gl-460:i386` and install the 32 bit userspace with something like `sudo apt install libnvidia-gl-460:i386`.

**Comment by foley2k2** [comment]
$ apt policy libnvidia-gl-460 libnvidia-gl-460:i386
libnvidia-gl-460:
  Installed: 460.84-0ubuntu0~0.20.04.1
  Candidate: 460.84-0ubuntu0~0.20.04.1
  Version table:
 *** 460.84-0ubuntu0~0.20.04.1 500
        500 http://ppa.launchpad.net/graphics-drivers/ppa/ubuntu focal/main amd64 Packages
        100 /var/lib/dpkg/status
     460.80-0ubuntu0.20.04.2 500
        500 http://archive.ubuntu.com/ubuntu focal-updates/restricted amd64 Packages
        500 http://security.ubuntu.com/ubuntu focal-security/restricted amd64 Packages
libnvidia-gl-460:i386:
  Installed: 460.84-0ubuntu0~0.20.04.1
  Candidate: 460.84-0ubuntu0~0.20.04.1
  Version table:
 *** 460.84-0ubuntu0~0.20.04.1 500
        500 http://ppa.launchpad.net/graphics-drivers/ppa/ubuntu focal/main i386 Packages
        100 /var/lib/dpkg/status
     460.80-0ubuntu0.20.04.2 500
        500 http://archive.ubuntu.com/ubuntu focal-updates/restricted i386 Packages
        500 http://security.ubuntu.com/ubuntu focal-security/restricted i386 Packages

$ sudo apt install libnvidia-gl-460:i386.
[sudo] password for *****:           
Reading package lists... Done
Building dependency tree       
Reading state information... Done
E: Unable to locate package libnvidia-gl-460:i386.
E: Couldn't find any package by glob 'libnvidia-gl-460:i386.'

**Comment by foley2k2** [comment]
Fixed it - driver was in a bad state and a reboot was needed. 
Recently, an update required that I roll back the driver.

Leaving this up there in case the minidump can be useful.
